include/ccf/kv/map_handle.h

```diff
@@ -229,16 +229,13 @@ namespace kv
 
     /** Delete a key-value pair.
      *
-     * It is safe to call this on non-existent keys - it will simply return
-     * false.
+     * It is safe to call this on non-existent keys.
      *
      * @param key Key to be removed
-     *
-     * @return true if the key had a value previously
      */
-    bool remove(const K& key)
+    void remove(const K& key)
     {
-      return write_handle.remove(KSerialiser::to_serialised(key));
+      write_handle.remove(KSerialiser::to_serialised(key));
     }
```

Comment:
@achamayou @MahatiC @jumaffre: This is a late-breaking, breaking-change - removing the return value from calls to `remove()`. The problem is that if we return a bool indicating whether this previously had a value, we may be introducing an implicit read dependency. ie, an application could do:

```
auto handle_foo = kv.rw(foo);
auto handle_bar = kv.rw(bar);
if (handle_foo->remove("A"))
{
  handle_bar->put("result", "Successfully removed A");
}
else
{
  handle_bar->put("result", "Failed to remove A");
}
```

For this to serialise correctly, we'd need to record and conflict a read dependency on all remove ops as well. Instead, I think it's simpler to remove this largely-unused return value. That makes `remove()` a pure-write, consistent with `put()` (which has no return value), and where apps actually _want_ a dependency like above to branch on, they can explicitly introduce it with a call to `has()`.

Another way to look at this is that it makes it possible to create pure-delete ops, which may be 're-ordered' like other pure-write ops. With a return value, there's a class of operations that the app-developer intended to be re-orderable as pure writes, but we track as reads and introduce unnecessarily conflicts.

Reply:
That seems fine for 3.x (given appropriate mention in release notes), but definitely not for backporting to 2.x.

Reply:
Agreed. I've added a `CHANGELOG` entry, and wasn't intending to backport this to 2.x.

However, this has made me wonder how this interacts over multiple versions with ledgers that contain tombstones... I'll go add another test.

Reply:
> it's simpler to remove this largely-unused return value. 

Yup, that seems like a good idea and a clean way to handle this